cpp/include/Ice/ValueFactory.h

```diff
@@ -106,112 +84,7 @@ class ICE_CLASS(ICE_API) ValueFactoryManager
-     */
-    virtual ::Ice::ValueFactory find(const ::std::string& id) const noexcept = 0;
-};
-
-}
-
-/// \cond STREAM
-namespace Ice
-{
-
-}
-/// \endcond
-
-/// \cond INTERNAL
-namespace Ice
-{
-
-using ValueFactoryManagerPtr = ::std::shared_ptr<ValueFactoryManager>;
-
-}
-/// \endcond
-
-#else // C++98 mapping
-
-namespace Ice
-{
-
-class ValueFactory;
-/// \cond INTERNAL
-ICE_API LocalObject* upCast(ValueFactory*);
-/// \endcond
-typedef ::IceInternal::Handle< ValueFactory> ValueFactoryPtr;
-
-class ValueFactoryManager;
-/// \cond INTERNAL
-ICE_API LocalObject* upCast(ValueFactoryManager*);
-/// \endcond
-typedef ::IceInternal::Handle< ValueFactoryManager> ValueFactoryManagerPtr;
-
-}
-
-namespace Ice
-{
-
-/**
- * A factory for values. Value factories are used in several places, such as when Ice receives a class instance and
- * when Freeze restores a persistent value. Value factories must be implemented by the application writer and
- * registered with the communicator.
- * \headerfile Ice/Ice.h
- */
-class ICE_API ValueFactory : public virtual LocalObject
-{
-public:
-
-    typedef ValueFactoryPtr PointerType;
-
-    virtual ~ValueFactory();
-    ValueFactory() = default;
-    ValueFactory(const ValueFactory&) = default;
-    ValueFactory& operator=(const ValueFactory&) = default;
-
-    /**
-     * Create a new value for a given value type. The type is the absolute Slice type id, i.e., the id relative to the
-     * unnamed top-level Slice module. For example, the absolute Slice type id for an interface <code>Bar</code> in
-     * the module <code>Foo</code> is <code>"::Foo::Bar"</code>.
-     * Note that the leading "<code>::</code>" is required.
-     * @param type The value type.
-     * @return The value created for the given type, or nil if the factory is unable to create the value.
-     */
-    virtual ValuePtr create(const ::std::string& type) = 0;
-};
-
-/// \cond INTERNAL
-inline bool operator==(const ValueFactory& lhs, const ValueFactory& rhs)
-{
-    return static_cast<const LocalObject&>(lhs) == static_cast<const LocalObject&>(rhs);
-}
-
-inline bool operator<(const ValueFactory& lhs, const ValueFactory& rhs)
-{
-    return static_cast<const LocalObject&>(lhs) < static_cast<const LocalObject&>(rhs);
-}
-/// \endcond
-
-/**
- * A value factory manager maintains a collection of value factories. An application can supply a custom
- * implementation during communicator initialization, otherwise Ice provides a default implementation.
- * @see ValueFactory
- * \headerfile Ice/Ice.h
- */
-class ICE_API ValueFactoryManager : public virtual LocalObject
-{
-public:
-
-    typedef ValueFactoryManagerPtr PointerType;
-
-    virtual ~ValueFactoryManager();
-    ValueFactoryManager() = default;
-    ValueFactoryManager(const ValueFactoryManager&) = default;
-    ValueFactoryManager& operator=(const ValueFactoryManager&) = default;
+    virtual void add(ValueFactoryFunc factory, const ::std::string& id) = 0;
```

Comment:
That's the only API we use in cpp/. 

The ValueFactoryPtr API is used by other language mappings based on C++ like python.